pkg/flowkit/config/json/config.go

```diff
@@ -108,9 +121,25 @@ type Parser struct{}
 func NewParser() *Parser {
 	return &Parser{}
 }
+func emulatorIsDefault(emulator config.Emulator) bool {
+	if emulator.Name == config.DefaultEmulatorConfigName &&
+		emulator.ServiceAccount == config.DefaultEmulatorServiceAccountName &&
+		emulator.Port == config.DefaultEmulatorPort {
+		return true
+	}
+	return false
+}
 
 // Serialize configuration to raw.
 func (p *Parser) Serialize(conf *config.Config) ([]byte, error) {
+	if emulatorIsDefault(conf.Emulators[0]) {
+		defaultJsonConf := transformDefaultConfigToJSON(conf)
+		data, err := json.MarshalIndent(defaultJsonConf, "", "\t")
+		if err != nil {
+			return nil, err
+		}
+		return data, nil
+	}
```

Comment:
I don't think this is a good place for this, also not sure if the logic works, since if it gets the default emulator then it just serializes the emulators and returns that as the whole config, the other config values are not even returned. This should be moved to the `json/emulator.go` to function `transformEmulatorsToJSON` where you can check if emulator you are trying to add to `jsonEmulator` is default and if so don't add it.

Reply:
addressed